scripts/performance/benchmark

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+"""
+Use for benchmarking performance of other scripts. Provides data about
+time, memory use, cpu usage, network in, network out about the script ran in
+the form of a csv.
+
+
+Usage
+=====
+
+NOTE: Make sure you run ``pip install -r requirements.txt`` before running.
+
+To use the script, run::
+
+    ./benchmark "./my-script-to-run"
+
+
+If no ``--output-file`` was provided, the data will be saved to
+``performance.csv``
+"""
+import argparse
+import os
+import sys
+import subprocess
+import time
+
+import psutil
+
+
+# Determine the interface to track network IO depending on the platform.
+if sys.platform.startswith('linux'):
+    INTERFACE = 'eth0'
+elif sys.platform == 'darwin':
+    INTERFACE = 'en0'
+else:
+    # TODO: Add support for windows. This would require figuring out what
+    # interface to use on windows.
+    raise RuntimeError('Script cannot be run on %s' % sys.platform)
+
+
+def benchmark(args):
+    parent_pid = os.getpid()
+    child_p = run_script(args)
+    try:
+        # Benchmark the process where the script is being ran.
+        run_benchmark(child_p.pid, args.output_file, args.data_interval)
+    except KeyboardInterrupt:
+        # If there is an interrupt, then try to clean everything up.
+        proc = psutil.Process(parent_pid)
+        procs = proc.children(recursive=True)
+
+        for child in procs:
+            child.terminate()
+
+        gone, alive = psutil.wait_procs(procs, timeout=1)
+        for child in alive:
+            child.kill()
+        sys.exit(0)
+
+
+def run_script(args):
+    # Run the script in a separate process.
+    script_args = args.script.split(' ')
+    return subprocess.Popen(args.script, shell=True)
+
+
+def run_benchmark(pid, output_file, data_interval):
+    p = psutil.Process(pid)
+    previous_net = psutil.net_io_counters(pernic=True)[INTERFACE]
+    previous_time = time.time()
+
+    with open(output_file, 'w') as f:
+        while p.is_running():
+            if p.status() == psutil.STATUS_ZOMBIE:
+                p.kill()
+                break
+            # Sleep for the data interval.
```

Comment:
FWIW, these comments are distracting.  I think it's pretty clear that `time.sleep(data_internal)` is sleeping for the data_interval.

In general, I only prefer comments that explain non-obvious things about the code or *why* the code is written this particular way.